src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -40,28 +40,8 @@ namespace Microsoft.IdentityModel.Protocols
     /// </summary>
     /// <typeparam name="T">The type of <see cref="IDocumentRetriever"/>.</typeparam>
     [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1001:TypesThatOwnDisposableFieldsShouldBeDisposable")]
-    public class ConfigurationManager<T> : IConfigurationManager<T> where T : class
+    public class ConfigurationManager<T> : StandardConfigurationManager, IConfigurationManager<T> where T : StandardConfiguration
```

Comment:
I think this will be breaking the 'where T : StandardConfiguration.

Reply:
Makes sense; I've changed this back to what it was originally.

Reply:
According to the comments, the StandardConfiguration "Represents a generic metadata configuration which is applicable for both XML and JSON based configurations.".
1. Is there anything wrong if we name it "MetadataConfiguration" or "BaseConfiguration" or 'ConfigurationBase"? "Standard" seems vague to me.
2. I think abstract class is fine as I feel that more implementation can be added. For example, adding a property that returns the type (JSON, XML,...), or a method returning a SecurityKey from the SigningKeys collection.

Reply:
Thanks for your feedback Danny! :)

I think I'm going to go with BaseConfiguration as per our discussion with the rest of the team. I agree that it's much less vague!

Will need to think more about the abstract class vs. interface dilemma... I guess this really depends on the likelihood of these two classes having any kind of default implementations.